src/hooks/useSessionWithWebApp.ts

```diff
@@ -0,0 +1,149 @@
+import { IShapeDiverExportDefinition } from "types/shapediver/export";
+import { IShapeDiverParameterDefinition } from "types/shapediver/parameter";
+
+/** Type used for parameter definitions */
+type IWebAppParameterDefinition = IShapeDiverParameterDefinition;
+
+/** Type used for export definitions */
+type IWebAppExportDefinition = IShapeDiverExportDefinition;
+
+/** Reference to a parameter (custom or defined by the session) */
+interface IWebAppParameterRef {
+	/** Optional id of the referenced parameter. Takes precedence over name. Id or name must be specified. */
+	id?: string
+	/** Optional name of the referenced parameter. Id takes precedence. Id or name must be specified. */
+	name?: string
+	/** Properties of the parameter to be overridden. */
+	overrides?: Pick<IWebAppParameterDefinition, "displayname" | "group" | "order" | "tooltip" | "hidden">
+	/** Disable the UI element of the parameter if its state is dirty. */
+	disableIfDirty: boolean
+	/** Ask the user to accept or reject changes of this parameter before executing them. */
+	acceptRejectMode: boolean
+}
+
+/** Reference to an export (defined by the session) */
+interface IWebAppExportRef {
+	/** Optional id of the referenced export. Takes precedence over name. Id or name must be specified. */
+	id?: string
+	/** Optional name of the referenced export. Id takes precedence. Id or name must be specified. */
+	name?: string
+	/** Properties of the export to be overridden. */
+	overrides?: Pick<IWebAppExportDefinition, "displayname" | "group" | "order" | "tooltip" | "hidden">
+}
+
+/** Types of widgets */
+type WebAppWidgetType = "accordion" | "text" | "image";
+
+/** 
+ * Properties of a parameter and export accordion widget.
+ * UI elements of the referenced parameters and exports are grouped 
+ * and ordered according to their properties (which might be overridden).
+ */
+interface IWebAppWidgetPropsAccordion {
+	/** References to parameters which shall be displayed by the accordion. */
+	parameters: IWebAppParameterRef[]
+	/** References to exports which shall be displayed by the accordion. */
+	exports: IWebAppExportRef[]
+}
+
+/** Properties of a text widget. */
+interface IWebAppWidgetPropsText {
+	/** Plain text. Takes precedence. */
+	text?: string
+	/** Optional markdown. */
+	markdown?: string
+}
+
+/** Properties of an image widget. */
+interface IWebAppWidgetPropsImage {
+	/** Base 64 encoded image. Takes precedence over export reference. */
+	data?: string
+	/** Optional reference to export which provides the image */
+	export?: IWebAppExportRef
+}
+
+/** A widget. */
+interface IWebAppWidget {
+	/** Type of the widget. */
+	type: WebAppWidgetType
+	/** Properties of the widget. */
+	props: IWebAppWidgetPropsAccordion | IWebAppWidgetPropsText | IWebAppWidgetPropsImage
+}
+
+/** 
+ * A tab displayed in a container.
+ */
+interface IWebAppContainerTab {
+	/** Name of the tab. */
+	name: string
+	/** Widgets displayed in the tab. */
+	widgets: IWebAppWidget[]
```

Comment:
@snabela It looks to me that we either need a widget type for simple parameters, otherwise parameters need to be contained in an accordion with the current specification.

Reply:
@mathieu-shapediver agreed, you mean we need a way to show parameter components without being grouped, did I understand correctly? 

We can add options to the accordion component, which allows us to configure things like: 
  * do not show accordion in case there is only one group
  * ignore groups
  * allow to open multiple groups at once

Related mantine doc: https://mantine.dev/core/accordion/

Reply:
@snabela I think a last change would make sense: I think text and image components should be defined at the same level as parameters and exports, so we can include them in accordions as well. With the previous change you did where it is now easier to also define parameters and exports, this would provide maximum flexibility.

Reply:
@mathieu-shapediver it is already possible to interleave text and image widgets with accordion widgets. If we want to support some kind of text and image widgets inside accordions, then those would need to be defined as pseudo parameters, which would need to have the same basic properties like parameters and exports (id, name, displayname, order, group, hidden, type). The type of these parameters could be `Label` or similar. 
I suggest to implement this addition as a separate task.